status-im/status-desktop
# Test(pytest) start aut

Fixes ([Prepare structure for "squish driver"](https://github.com/status-im/desktop-qa-automation/issues/67))
Fixes ([Implement managing squishserver during a test](https://github.com/status-im/desktop-qa-automation/issues/68))
Fixes ([Implement setup, start, and stop the application with custom user data](https://github.com/status-im/desktop-qa-automation/issues/69))
Fixes ([Prepare the Main Window fixture](https://github.com/status-im/desktop-qa-automation/issues/70))
Fixes ([Save screenshot on failed test](https://github.com/status-im/desktop-qa-automation/issues/71))

### Manage squish server and AUT

- Wrappers for UI elements moved from the previous version framework
- Methods for starting/stopping the squish server added
- Methods for launching, stopping, and detaching the application added

I found, for some versions of PyCharm, an environment variable, which we add (SQUISH_DIR, PYTHONPATH, LD_LIBRARY_PATH) replaced by PyCharm, so more stable way to add environment variable, to set it specifically for the test:
![image](https://github.com/status-im/status-desktop/assets/128374224/2239cd9d-f9ab-4f01-8513-7a0623a974b4)

**Before testing, please update the requirements: step 6 in the README file**

Test (execute for CI app build)
1)Set a breakpoint after driver.context.detach() in test_start_aut and execute the test in debug mode.
![image](https://github.com/status-im/status-desktop/assets/128374224/1d792dde-3f2a-4f0a-a896-5fc03c63df06)

2)When the execution stops on the breakpoint, open squish IDE, and open suite settings
3) add Attachable AUT in 'Manage AUT' with the parameters:
- Name: Status
- Host: localhost
- Port: 61500
<img width="659" alt="image" src="https://github.com/status-im/status-desktop/assets/128374224/4a05ddf1-2df0-45ef-a34f-b174b1818dbf">

4) Select 'No Application' in AUT settings
5) Save changes and close suite settings.
6) Start AUT by button 
<img width="26" alt="image" src="https://github.com/status-im/status-desktop/assets/128374224/6210a9c4-4c33-4e6d-bbda-389a88836499">
on the toolbar:
<img width="382" alt="image" src="https://github.com/status-im/status-desktop/assets/128374224/6b8b8ead-2eca-4400-b245-026babd36cb2">

7) In the 'Launch Settings' window select attachable Status and click  <OK> button
<img width="286" alt="image" src="https://github.com/status-im/status-desktop/assets/128374224/4d39724c-4a4c-4eab-bce7-b8a0c6d006f6">

Elements should be visible for the Pick tool in Squish IDE

We have now two methods to start the test application:
- by subprocesses (we will use it only for local run, to explore application inside)
- by squish startApplication (attach/detach mode not available)
Start AUT method can be selected by ATTACH_MODE key in ._local.py file
